### PR TITLE
CONTRIBUTING: offer clearer guidelines about blocker vs. non-blocker issues

### PR DESCRIPTION
**Pull request by RaitoBezarius**
## Description of changes

Brought by https://matrix.to/#/!kjdutkOsheZdjqYmqp:nixos.org/$IuFJ-rB798l9jYfcWm9RyYfD5EDb420xc0kwa-FgZZo?via=nixos.org&via=matrix.org&via=nixos.dev discussion.

We sometimes have reviews that are blocked because of miscommunication between author and reviewer, we make this clear so that reviewers and authors can refer to a block of "authoritative" text to unblock their own situations.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -316,6 +316,12 @@ When reviewing a pull request, please always be nice and polite. Controversial c
 
 GitHub provides reactions as a simple and quick way to provide feedback to pull requests or any comments. The thumb-down reaction should be used with care and if possible accompanied with some explanation so the submitter has directions to improve their contribution.
 
+When doing a review, consider focusing on the changes proposed, if you find out about anything that could be improved but is not part of the proposed changes, consider tracking it as an issue, **offer** the original contributor to do a follow-up pull request or do it in that pull request, but remember that you should ideally not make this a blocker or give the impression to the contributor this would be a blocker for this merge. If the follow-up change is structural for the contribution, make your point clear about why you feel it is important to do it before the proposed changes.
+
+Also, consider giving to the contributor a clear representation of what you consider to be blocker and not to help them prioritize their efforts towards a potential merge.
```
Comment (infinisil) [inline CONTRIBUTING.md:321]:
```suggestion
Also, consider giving to the contributor a clear representation of what you consider to be blocker and to help them prioritize their efforts towards a potential merge.
```

?

**Review by infinisil — DISMISSED** [review]
This section is quite large already, would be nice to have this new part (or all of its parts really) in sub sections. This way they can also be linked to.

**Comment by kirillrdy** [comment]
don't want to increase the scope of this PR, but how about something like https://conventionalcomments.org/ ? so that suggestions during review are very obviously different from requests

File: CONTRIBUTING.md (hunk 1 above)
Comment (FliegendeWurst) [inline CONTRIBUTING.md:321]:
```suggestion
Also, consider giving the contributor a clear representation of what you consider to be blockers to help them prioritize their efforts towards a potential merge.
```

**Comment by RaitoBezarius** [comment]
> don't want to increase the scope of this PR, but how about something like [conventionalcomments.org](https://conventionalcomments.org/) ? so that suggestions during review are very obviously different from requests

I'm okay with mentioning as an example.

**Pull request merged**